### PR TITLE
[WIP] insert bailouts on frontiers of fusion groups

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 1]
```diff
@@ -146,6 +148,13 @@ bool allShapesAreKnown(Value* v) {
   return v->isCompleteTensor();
 }
 
+bool allShapesAreKnown2(Value* v) {
+  if (!v->type()->cast<TensorType>()) {
+    return true;
+  }
+  return v->isCompleteTensor();
```
Comment (eellison) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp:155]: why are we checking that the tensor is complete ? aren't we also expecting tensor types which are profiled but not fully known ?

File: torch/csrc/jit/runtime/profiling_record.cpp [hunk 2]
```diff
@@ -176,6 +176,16 @@ bool needsProfiledInputs(Node* n) {
   }
 }
 
+bool inputsProfilable(Node* n) {
+  for (auto iv : n->inputs()) {
+    if (!needsProfiledInputs(iv->node()) && n->kind() != prim::Constant) {
```
Comment (eellison) [inline torch/csrc/jit/runtime/profiling_record.cpp:181]: constants don't have any inputs so you shouldn't need to add the `&& n->kind() != prim::Constant` case. or maybe you meanat `iv->node()` ?

**Review by eellison — COMMENTED** [review]
So the idea here is if we expect a node's inputs to be completely profiled than we do not need to profile the node itself ? Do you mind adding a test and also filling out the summary when you want to make this no longer a WIP.

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp (hunk 1 above)
Comment (Krovatkin) [reply torch/csrc/jit/passes/tensorexpr_fuser.cpp:155]: NNC doesn't know how to handle those yet

File: torch/csrc/jit/runtime/profiling_record.cpp (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/profiling_record.cpp:181]: yes, this is wrong. We should just skip over constants. Let me fix this

**Pull request closed without merging**